src/agent/TagAgent.tsx

```diff
@@ -1,9 +1,17 @@
 import { ChatOpenAI } from '@langchain/openai';
 import { ChatPromptTemplate } from '@langchain/core/prompts';
-import { JsonOutputParser } from '@langchain/core/output_parsers';
+import { StructuredOutputParser } from '@langchain/core/output_parsers';
+import { z } from 'zod';
+import { RunnableSequence } from '@langchain/core/runnables';
 
 const openaiApiKey = import.meta.env.VITE_OPENAI_API_KEY;
 
+const schema = z.object({
+  tags: z.array(z.string()),
+});
+
+const parser = StructuredOutputParser.fromZodSchema(schema);
+
```

Comment:
```suggestion
const schema = { foo: "bar" };

const modelWithStructure = model.withStructuredOutput(schema);
```

이런 방식으로 모델 뒤에 `withSturcturedOutput` 메서드를 끼얹어서.. 구조화된 출력을 정의해주고

Reply:
오.. 이런 방식도 있군요... 많이 배워갑니다 ㅎㅎ